drivers/display/display_st7701.c

```diff
@@ -0,0 +1,449 @@
+	uint8_t gip_e1[12];
+	uint8_t gip_e2[14];
+	uint8_t gip_e3[5];
+	uint8_t gip_e4[3];
+	uint8_t gip_e5[17];
+	uint8_t gip_e6[5];
+	uint8_t gip_e7[3];
+	uint8_t gip_e8[17];
+	uint8_t gip_eb[8];
+	uint8_t gip_ec[3];
+	uint8_t gip_ed[17];
+	uint8_t pvgamctrl[17];
+	uint8_t nvgamctrl[17];
+};
+
+struct st7701_data {
+	uint16_t xres;
+	uint16_t yres;
+	uint8_t dsi_pixel_format;
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+};
+
+static inline int st7701_dcs_write(const struct device *dev, uint8_t cmd, const void *buf,
+				   size_t len)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	ret = mipi_dsi_dcs_write(cfg->mipi_dsi, cfg->channel, cmd, buf, len);
+	if (ret < 0) {
+		LOG_ERR("DCS 0x%x write failed! (%d)", cmd, ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7701_short_write_1p(const struct device *dev, uint8_t cmd, uint8_t val)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+	uint8_t buf[] = {cmd, val};
+
+	ret = mipi_dsi_generic_write(cfg->mipi_dsi, cfg->channel, buf, sizeof(val));
+	if (ret < 0) {
+		LOG_ERR("Short write failed! (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7701_generic_write(const struct device *dev, const void *buf, size_t len)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	ret = mipi_dsi_generic_write(cfg->mipi_dsi, cfg->channel, buf, len);
+	if (ret < 0) {
+		LOG_ERR("Generic write failed! (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7701_check_id(const struct device *dev)
+{
+	const struct st7701_config *cfg = dev->config;
+	uint32_t id = 0;
+	int ret;
+
+	ret = mipi_dsi_dcs_read(cfg->mipi_dsi, cfg->channel, ST7701_CMD_ID1, &id, sizeof(id));
+	if (ret != sizeof(id)) {
+		LOG_ERR("Read panel ID failed! (%d)", ret);
+		return -EIO;
+	}
+
+	if (id != ST7701_ID) {
+		LOG_ERR("ID 0x%x (should 0x%x)", id, ST7701_ID);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int st7701_configure(const struct device *dev)
+{
+	struct st7701_data *data = dev->data;
+	const struct st7701_config *cfg = dev->config;
+	uint8_t buf[4];
+	int ret;
+
+	const uint8_t control0[] = {DSI_CMD2BKX_SEL, 0x77, 0x01, 0x00, 0x00, DSI_CMD2BK0_SEL};
+	const uint8_t control1[] = {0xC0, 0x63, 0x00};
+	const uint8_t control2[] = {0xC1, 0x11, 0x02};
+	const uint8_t control3[] = {0xC2, 0x01, 0x08};
+	const uint8_t control4[] = {0xCC, 0x18};
+
+	st7701_generic_write(dev, control0, sizeof(control0));
+	st7701_generic_write(dev, control1, sizeof(control1));
+	st7701_generic_write(dev, control2, sizeof(control2));
+	st7701_generic_write(dev, control3, sizeof(control3));
+	st7701_generic_write(dev, control4, sizeof(control4));
+
+	/* Gamma Cluster Setting */
+	st7701_generic_write(dev, cfg->pvgamctrl, sizeof(cfg->pvgamctrl));
+	st7701_generic_write(dev, cfg->nvgamctrl, sizeof(cfg->nvgamctrl));
+
+	/* Initial power control registers */
+	const uint8_t _FF1[] = {DSI_CMD2BKX_SEL, 0x77, 0x01, 0x00, 0x00, DSI_CMD2BK1_SEL};
```

Comment:
Variable definition should be at the top of the block. Variable name should be in lower case. Must not be prefixed with `_`.